File: code projet/Web_3A_G7/Rendu_Web/Web2/modele/utilisateur.php
<?php
require_once("modele.php");

class Utilisateur extends Modele {
    protected static $objet = "utilisateur";
    protected static $cle = "id_utilisateur";

    public function __construct($param = NULL) {
        if (is_array($param)) {
            parent::__construct($param);
        }
    }

    public function login($identifiant, $mdp) {
        $req = "SELECT * FROM Utilisateur WHERE identifiant_connexion = :login";
        $stmt = Connexion::pdo()->prepare($req);
        $stmt->execute(['login' => $identifiant]);
        $user = $stmt->fetch(PDO::FETCH_ASSOC);

        if ($user && password_verify($mdp, $user['hash_mdp'])) {
            $token = bin2hex(random_bytes(16));
            return [
                "token" => $token,
                "utilisateur" => [
                    "idUtilisateur" => $user['id_utilisateur'],
                    "identifiantConnexion" => $user['identifiant_connexion'],
                    "role" => $user['code_role']
                ]
            ];
        }
        return false;
    }
}
?>

File: code projet/COO_3A_G7/JAVA2/ApplicationJava/src/modele/Utilisateur.java
package modele;

public class Utilisateur {
    protected int idUtilisateur;
    protected String identifiantConnexion;
    protected String hashMdp;
    protected Role role;

    public Utilisateur() {}

    public boolean seConnecter() {
        return this.idUtilisateur > 0;
    }

    public int getIdUtilisateur() { return idUtilisateur; }
    public void setIdUtilisateur(int idUtilisateur) { this.idUtilisateur = idUtilisateur; }
    public String getIdentifiantConnexion() { return identifiantConnexion; }
    public void setIdentifiantConnexion(String identifiantConnexion) { this.identifiantConnexion = identifiantConnexion; }
    public String getHashMdp() { return hashMdp; }
    public void setHashMdp(String hashMdp) { this.hashMdp = hashMdp; }
    public Role getRole() { return role; }
    public void setRole(Role role) { this.role = role; }


}